crypto/err/err.c

```diff
@@ -265,33 +283,33 @@ static void build_SYS_str_reasons(void)
 }
 #endif
 
-#define err_clear_data(p, i) \
-        do { \
-            if ((p)->err_data_flags[i] & ERR_TXT_MALLOCED) {\
-                OPENSSL_free((p)->err_data[i]); \
-                (p)->err_data[i] = NULL; \
-            } \
-            (p)->err_data_flags[i] = 0; \
-        } while (0)
-
-#define err_clear(p, i) \
-        do { \
-            err_clear_data(p, i); \
-            (p)->err_flags[i] = 0; \
-            (p)->err_buffer[i] = 0; \
-            (p)->err_file[i] = NULL; \
-            (p)->err_line[i] = -1; \
-        } while (0)
+static void err_clear_data(ERR_STATE *s, int i)
+{
+    ERR_ENTRY *p = &s->entries[i];
+
+    if (p->err_data_flags & ERR_TXT_MALLOCED) {
```

Comment:
there are are a few of these '&' tests in this PR? Are they supposed to be (flags & X) != 0